frontend/components/search.js

```diff
@@ -68,22 +78,27 @@ class Search {
 
     if (results.error) {
       macros.error('Error with networking request', results.error);
-      return [];
+      return { results: [] };
     }
 
-
+    // if cache doesn't exist, instantiate. Subject info only changed here
+    // since it should only be changed on cache misses
     if (!this.cache[termId + query]) {
-      this.cache[termId + query] = [];
+      this.cache[termId + query] = {};
+      this.cache[termId + query].results = [];
+      this.cache[termId + query].subjectName = waitedRequest.subjectName;
+      this.cache[termId + query].subjectCount = waitedRequest.subjectCount;
```

Comment:
oh and can you change this to this?

```js
this.cache[termId + query] = {
  results: []
  subjectName: waitedRequest.subjectName, 
  subjectCount: waitedRequest.subjectCount,
}
```

Reply:
whoah getting rid of redundant code hwat